scripts/fetch-configlet

```diff
@@ -45,14 +45,31 @@ get_download_url() {
     cut -d'"' -f4
 }
 
-download_url="$(get_download_url)"
-output_dir="bin"
-output_path="${output_dir}/latest-configlet.${ext}"
-curl "${curlopts[@]}" --output "${output_path}" "${download_url}"
-
-case "${ext}" in
-  *zip) unzip "${output_path}" -d "${output_dir}"   ;;
-  *)    tar xzf "${output_path}" -C "${output_dir}" ;;
-esac
+# Check if we are in the root of repo that has a $REPO_ROOT/bin
+# This is needed as `curl` will output to `$REPO_ROOT/bin/`
+in_repo () {
+  local root
+  root=$(git rev-parse --show-toplevel 2>/dev/null) || return
+  [[ "${PWD}" = "${root}"/* ]] && [[ -d "${root}/bin" ]]
+}
+
+main () {
```

Comment:
What was the reason to introduce a `main` function instead of the old approach?

Reply:
@ErikSchierboom So that we can have the `return 1` line.

If we remove the `main` function then we get the error
```
return: can only `return' from a function or sourced script
```


But I think the script is long enough that `main` improves readability anyway. 

See also https://google.github.io/styleguide/shellguide.html#s7.8-main

Reply:
> So that we can have the return 1 line.

Right. `exit 1` would also have worked, right?

> But I think the script is long enough that main improves readability anyway.

🤷 I'm okay with it.

Reply:
What ee7 said. The main function, like in many languages, doesn't alter behavior as much as serve to group a logical unit of code. The top level was getting crowded. A main function isolates the "main" logic.